jcl/src/java.base/share/classes/java/lang/invoke/VarargsCollectorHandle.java

```diff
@@ -170,7 +170,12 @@ public MethodHandle asType(MethodType newType) throws ClassCastException {
 		}
 		CollectHandle collector = previousCollector;
 		if ((collector == null) || (collector.collectArraySize != collectCount)) {
-			collector = (CollectHandle) next.asCollector(arrayType, collectCount);
+			try {
+				collector = (CollectHandle) next.asCollector(arrayType, collectCount);
+			} catch (IllegalArgumentException iae) {
+				/*[MSG "K0681", "Failed to build collector"]*/
+				throw new WrongMethodTypeException(com.ibm.oti.util.Msg.getString("K0681"), iae); //$NON-NLS-1$
```

Comment:
Can the exception be moved to a helper method?

```java

private WrongMethodTypeException throwNewWMTE(IlegalArgumentException iae) {
  throw new WrongMethodTypeException(com.ibm.oti.util.Msg.getString("K0681"), iae); //$NON-NLS-1$
}

```
and the call becomes the code below to make it clear to the reader (and the JIT!) that there is no path past this code:
```
throw throwNewWMTE(iae);
```

Reply:
Sure, PR has been updated accordingly and verified manually that the test passes with this PR.

Reply:
Can you push the change again?

Reply:
Oops, sorry, it should be updated now.